packages/rs-drive-proof-verifier/src/proof/from_proof.rs

```diff
@@ -69,15 +70,15 @@ pub trait FromProof<Req> {
     /// * `Ok(object)` when the requested object was found in the proof.
     /// * `Err(Error::DocumentMissingInProof)` when the requested object was not found in the proof.
     /// * `Err(Error)` when either the provided data is invalid or proof validation failed.
-    fn from_proof(
-        request: &Req,
-        response: &Self::Response,
-        provider: Box<dyn QuorumInfoProvider>,
+    fn from_proof<'a, I: Into<Self::Request>, O: Into<Self::Response>>(
+        request: I,
+        response: O,
+        provider: &'a dyn QuorumInfoProvider,
```

Comment:
Not a big fan of &'a dyn QuorumInfoProvider, but we can leave it for now.

Reply:
Already explained, it's needed if we want to make C bindings and provide QuorumInfoProvider from C code.